kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -434,7 +435,11 @@ private String getCommitFilename(String encodedPartition) {
   }
 
   private String fileKey(String topicsPrefix, String keyPrefix, String name) {
-    return topicsPrefix + dirDelim + keyPrefix + dirDelim + name;
+    String base = "";
+    if (StringUtils.isNotBlank(topicsPrefix) && !topicsPrefix.equals("/")) {
```

Comment:
Seems we might be overly specific by allowing everything besides `/`. If someone set their `topics.dir` property to `/` they might know what they're doing (In S3 this produces a different key than the empty string obviously, but I don't it should be on us to prevent this here).